src/server/_query.py

```diff
@@ -174,29 +174,25 @@ def filter_pair(pair: SourceSignalPair, i) -> str:
 def filter_time_pairs(
     type_field: str,
     time_field: str,
-    values: Sequence[TimePair],
+    pair: Optional[TimePair],
     param_key: str,
     params: Dict[str, Any],
 ) -> str:
     """
-    returns the SQL sub query to filter by the given time pairs
+    returns the SQL sub query to filter by the given time pair
     """
+    if not pair:
+        return "FALSE"
```

Comment:
Question: is this a good code path to have? AFAICT there's an exception raised in `covidcast.py:parse_time_pairs` if you don't have either: a) `time=...` or b) `time_type=...` and `time_values=...`. So this will technically never happen in the API. So:
- Pro: function safety, can be reused in the future in other places that didn't have the same validation
- Con: confusing now, because it makes you think this is a possible code path

not super blocking, i can go either way, but curious what everyone thinks here, cc @melange396 @krivard

Reply:
Like you've mentioned, I'd rather have the extra failsafe for function safety so I kept this for now (but it should be trivial to remove if needed)